LanguageFeatures/Inline-classes/dynamic_semantics_inline_class_A03_t02.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A type test, o is U or o is! U, and a type cast, o as U, where U
+/// is or contains an inline type, is performed at run time as a type test and
+/// type cast on the run-time representation of the inline type
+///
+/// @description Check that at run time a type cast, `o as U` is performed as a
+/// type cast on the run-time representation of the inline type
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/expect.dart";
+
+inline class V {
+  final int id;
+  V(this.id);
+
+  String get foo => "Inline";
+}
+
+main() {
+  var v = V(42);
+  Expect.isTrue((v as int).isEven);
```

Comment:
Here we'd like to know that `_ as V` works the same as `_ as int` (or, perhaps use a representation type which is `List<String>`, just to get the variation; in that case `_ as V` should work the same as `_ as List<String>`).

"they both throw, or both of them don't throw" is less convenient than "they are equal", but it should be expressible using a try/catch and some boolean variables.